### PR TITLE
Revert "HADOOP-16822. Provide source artifacts for hadoop-client-api"

### PR DESCRIPTION
**Pull request by pan3793**
This reverts commit 2c4ab72a60113e4dd4ef2375e6f9413e519b1044.

### Description of PR

TL;DR, it generate the wrong source code and messes the code jump in IDEA.

See details in https://issues.apache.org/jira/browse/HADOOP-16822

### How was this patch tested?

Pass CI.

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by pan3793** [comment]
cc @jojochuang @steveloughran

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  60m  0s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  47m 43s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   0m 21s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  mvnsite  |   0m 25s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 25s |  |  trunk passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 23s |  |  trunk passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  shadedclient  |  86m 44s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 21s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 13s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 13s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  javac  |   0m 13s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 14s |  |  the patch passed with JDK Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   0m 13s |  |  the patch passed with JDK Private Build-1.8.0_392-8u392-ga-1~20.04-b08  |
| +1 :green_heart: |  shadedclient  |  38m  7s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 18s |  |  hadoop-client-api in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  |  The patch does not generate ASF License warnings.  |
|  |   | 197m 55s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6458/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6458 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint |
| uname | Linux 41ba3b906dea 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 34a7bca208b217efe2fbc0b35f4cab0363456fd4 |
| Default Java | Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.21+9-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_392-8u392-ga-1~20.04-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6458/1/testReport/ |
| Max. process+thread count | 566 (vs. ulimit of 5500) |
| modules | C: hadoop-client-modules/hadoop-client-api U: hadoop-client-modules/hadoop-client-api |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6458/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by pan3793** [comment]
Also cc @pjfanning @Hexiaoqiao

**Comment by pan3793** [comment]
I think we already reached the consensus in the JIRA discussion, could any committer take a look?

cc @slfan1989 @sunchao

**Review by cxzl25 — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
cc @aajisaka @steveloughran

**Review by steveloughran — APPROVED** [review]

**Comment by steveloughran** [comment]
done, though as I pressed sent I realised the title had pulled line #2 up; sorry.

will cherrypick to 3.4

**Comment by steveloughran** [comment]
merged to 3.4; branch-3.3 had conflicts. @pan3793 if you want it there, create another PR so yetus can validate it. thanks!
